docs/en/api-reference/system/ulp.rst

```diff
@@ -50,6 +50,21 @@ To compile the ULP FSM code as part of the component, the following steps must b
     ulp_embed_binary(${ulp_app_name} "${ulp_s_sources}" "${ulp_exp_dep_srcs}")
 
 The first argument to ``ulp_embed_binary`` specifies the ULP FSM binary name. The name specified here will also be used by other generated artifacts such as the ELF file, map file, header file and linker export file. The second argument specifies the ULP FSM assembly source files. Finally, the third argument specifies the list of component source files which include the header file to be generated. This list is needed to build the dependencies correctly and ensure that the generated header file will be created before any of these files are compiled. See the section below for the concept of generated header files for ULP applications.
+Variables in the ULP code will be prefixed with ``ulp_`` (default value) in this generated header file.
+
+If you need to embed multiple ULP program, in order to avoid conflicting variable names, it's possible to modify the prefix like this:
```

Comment:
Nitpick:

```suggestion
If you need to embed multiple ULP programs, you may add a custom prefix in order to avoid conflicting variable names like this:
```